cpp/src/rolling/rolling_detail.cuh

```diff
@@ -1020,10 +1020,12 @@ __launch_bounds__(block_size) __global__
     int64_t following_window = following_window_begin[i];
 
     // compute bounds
-    auto start = static_cast<size_type>(
-      min(static_cast<int64_t>(input.size()), max(0L, i - preceding_window + 1)));
-    auto end = static_cast<size_type>(
-      min(static_cast<int64_t>(input.size()), max(0L, i + following_window + 1)));
+    auto start =
+      static_cast<size_type>(min(static_cast<int64_t>(input.size()),
+                                 max(0L, static_cast<cudf::size_type>(i) - preceding_window + 1)));
+    auto end =
+      static_cast<size_type>(min(static_cast<int64_t>(input.size()),
+                                 max(0L, static_cast<cudf::size_type>(i) + following_window + 1)));
```

Comment:
It seems like all the math here is being done in `int64_t` except for the cast of `i`, which is to make it a signed value. Using `int64_t` here makes the expression self-consistent. See also: other comment about `gpu_rolling_new`, which does not use `int64_t` math.
```suggestion
    auto start =
      static_cast<size_type>(min(static_cast<int64_t>(input.size()),
                                 max(0L, static_cast<int64_t>(i) - preceding_window + 1)));
    auto end =
      static_cast<size_type>(min(static_cast<int64_t>(input.size()),
                                 max(0L, static_cast<int64_t>(i) + following_window + 1)));
```